pkg/clustertree/cluster-manager/controllers/mcs/auto_mcs_controller.go

```diff
@@ -124,15 +134,19 @@ func (c *AutoCreateMCSController) SetupWithManager(mgr manager.Manager) error {
 	)
 
 	shouldEnqueue := func(obj client.Object) bool {
-		annotations := obj.GetAnnotations()
-		if annotations == nil {
-			return false
+		if c.AutoCreateMCSPrefix != "" {
+			if strings.HasPrefix(obj.GetNamespace(), c.AutoCreateMCSPrefix) {
+				return true
+			}
 		}
-		if _, exists := annotations[utils.AutoCreateMCSAnnotation]; exists {
-			return true
-		} else {
-			return false
+
+		annotations := obj.GetAnnotations()
+		if annotations != nil {
+			if _, exists := annotations[utils.AutoCreateMCSAnnotation]; exists {
+				return true
+			}
 		}
+		return false
 	}
```

Comment:
line 142-line149  can reuse the func ```hasAutoMCSAnnotation```  
use ```obj.(*corev1.Service)``` to convert client.Object to corev1.Service